calico_mesos/calico_mesos.py

```diff
@@ -507,8 +507,13 @@ def _allocate(num_ipv4, num_ipv6, hostname, uid):
         "error": None  # Not None indicates error and contains error message.
     }
     """
-    result = datastore.auto_assign_ips(num_ipv4, num_ipv6, uid, {},
-                                       hostname=HOSTNAME)
+    try:
+        result = datastore.auto_assign_ips(num_ipv4, num_ipv6, uid, {},
+                                           hostname=HOSTNAME)
```

Comment:
It seems strange to me that we're seeing the host affinity error here. All we're providing is a hostname and num_ipv4. Libcalico should pick a block which that host has affinity to and use it. 

Is this in the wrong place, and we should be putting it around the isolate function calls?